larreco/RecoAlg/ProjectionMatchingAlg.cxx

```diff
@@ -0,0 +1,1151 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+// Class:       ProjectionMatchingAlg
+// Author:      D.Stefan (Dorota.Stefan@ncbj.gov.pl) and R.Sulej (Robert.Sulej@cern.ch), May 2015
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include "larreco/RecoAlg/ProjectionMatchingAlg.h"
+
+#include "larcore/CoreUtils/ServiceUtil.h" // lar::providerFrom<>()
+#include "lardata/DetectorInfoServices/DetectorPropertiesService.h"
+#include "larreco/RecoAlg/PMAlg/Utilities.h"
+
+#include "larevt/CalibrationDBI/Interface/ChannelStatusService.h"
+#include "larevt/CalibrationDBI/Interface/ChannelStatusProvider.h"
+
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+pma::ProjectionMatchingAlg::ProjectionMatchingAlg(const fhicl::ParameterSet& pset)
+  : fDetProp(lar::providerFrom<detinfo::DetectorPropertiesService>())
+{
+	this->reconfigure(pset); 
+}
+// ------------------------------------------------------
+
+pma::ProjectionMatchingAlg::~ProjectionMatchingAlg(void)
+{
+}
+// ------------------------------------------------------
+
+void pma::ProjectionMatchingAlg::reconfigure(const fhicl::ParameterSet& p)
+{
+	fOptimizationEps = p.get< double >("OptimizationEps");
+	fFineTuningEps = p.get< double >("FineTuningEps");
+
+	fTrkValidationDist2D = p.get< double >("TrkValidationDist2D");
+	fHitTestingDist2D = p.get< double >("HitTestingDist2D");
+
+	fMinTwoViewFraction = p.get< double >("MinTwoViewFraction");
+	
+	fDetProp = lar::providerFrom<detinfo::DetectorPropertiesService>();
+	
+	pma::Node3D::SetMargin(p.get< double >("NodeMargin3D"));
+
+	pma::Element3D::SetOptFactor(geo::kZ, p.get< double >("HitWeightZ"));
+	pma::Element3D::SetOptFactor(geo::kV, p.get< double >("HitWeightV"));
+	pma::Element3D::SetOptFactor(geo::kU, p.get< double >("HitWeightU"));
+}
+// ------------------------------------------------------
+
+double pma::ProjectionMatchingAlg::validate(const pma::Track3D& trk,
+	const std::vector< art::Ptr<recob::Hit> >& hits,
+	unsigned int testView) const
+{
+	double max_d = fTrkValidationDist2D;
+	double d2, max_d2 = max_d * max_d;
+	unsigned int nAll = 0, nPassed = 0;
+
+	std::vector< unsigned int > trkTPCs = trk.TPCs();
+	std::vector< unsigned int > trkCryos = trk.Cryos();
+	std::map< std::pair< unsigned int, unsigned int >, std::pair< TVector2, TVector2 > > ranges;
```

Comment:
Hi Robert,

  I hope that your trip went well.  See you and Dorota tomorrow.

  My plan is to write down every comment that comes to mind even fairly minor ones.  This way they don’t get lost.   We can worry about priorities later.  Of course some might be rejected.

      Rob

> On May 9, 2016, at 6:56 PM, Robert Sulej <notifications@github.com> wrote:
> 
> In larreco/RecoAlg/ProjectionMatchingAlg.cxx:
> 
> > +	pma::Element3D::SetOptFactor(geo::kV, p.get< double >("HitWeightV"));
> > +	pma::Element3D::SetOptFactor(geo::kU, p.get< double >("HitWeightU"));
> > +}
> > +// ------------------------------------------------------
> > +
> > +double pma::ProjectionMatchingAlg::validate(const pma::Track3D& trk,
> > +	const std::vector< art::Ptr<recob::Hit> >& hits,
> > +	unsigned int testView) const
> > +{
> > +	double max_d = fTrkValidationDist2D;
> > +	double d2, max_d2 = max_d * max_d;
> > +	unsigned int nAll = 0, nPassed = 0;
> > +
> > +	std::vector< unsigned int > trkTPCs = trk.TPCs();
> > +	std::vector< unsigned int > trkCryos = trk.Cryos();
> > +	std::map< std::pair< unsigned int, unsigned int >, std::pair< TVector2, TVector2 > > ranges;
> 
> Hello,
> Arrived. I can see this and following comments, thanks.
> R.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>